src/components/layout/header/header.tsx

```diff
@@ -161,6 +162,7 @@ const Header: React.FC<HeaderProps> = (props) => {
     <StyledHeader
       $lightTheme={Boolean(props.$lightTheme)}
       $transparent={isHeaderTransparent}
+      onKeyDown={(e) => e.keyCode === 27 && setIsNavigationVisible(false)}
```

Comment:
Please store the value `27` in a local variable `ESC_KEY_CODE` so it's speaking for itself, right now you don't know what's happening.